### ISSUE TITLE
[FEATURE] (Documentation) Having Real Difficulty Getting Started

### ISSUE THREAD

**Issue by chaojian-zhang** [fenix-hub/godot-engine.easy-charts#99]
**Describe the solution you'd like**
Thanks for the library! The functionalities seems promising and the examples are cool but I have difficulty setting things up from scratch.
In below I am referring to Godot 4.2

**Describe an implementation suggestion**
From the Wiki page https://github.com/fenix-hub/godot-engine.easy-charts/wiki/Static-and-Dynamic-data-plotting-with-PieChart it mentioned two functions `plot_from_csv()` and `plot_from_array()` - but none of those exist in the current code base at all.
Can you please update the wiki with a very basic Getting Started - in an empty Godot project, how does one add and make use of your library, e.g. to draw a simple line chart?
Thanks!

**Additional context**
* Godot 4.2
* Easy Chart 14.08.2023 (as downloaded from Asset Lib)

**Comment by chaojian-zhang** [comment]
In the meantime, I figured out a simple setup by utilizing examples, **though as examples uses a "control" scene, I am not sure whether it's the most efficient way to utilize this library**:

1. Create empty project
2. Use Asset Lib to download Easy Chart
3. Duplicate one of the examples to serve as main file
4. Essentially in a scene we just need to reference `res://addons/easy_charts/control_charts/chart.tscn` as a node/control
5. Then in the script we just do the follow for setting up the intended plot

```gdscript
extends Control

@onready var chart: Chart = $VBoxContainer/Chart

func _ready():
	# Y values can be n-size array of arrays.
	# x.size() == y.size() or`x.size() == y[n].size()
	var x: PackedFloat32Array = ArrayOperations.multiply_float(range(-10, 11, 1), 0.5)
	var y: Array = ArrayOperations.multiply_int(ArrayOperations.cos(x), 20)
	
	# More like chart properties, which can be distinct from trace properties
	var properties: ChartProperties = ChartProperties.new()
	properties.colors.frame = Color("#161a1d")
	properties.colors.background = Color.TRANSPARENT
	properties.colors.grid = Color("#283442")
	properties.colors.ticks = Color("#283442")
	properties.colors.text = Color.WHITE_SMOKE
	properties.draw_bounding_box = false
	properties.title = "Air Quality Monitoring"
	properties.x_label = "Time"
	properties.y_label = "Sensor values"
	properties.x_scale = 5
	properties.y_scale = 10
	properties.interactive = true # false by default, it allows the chart to create a tooltip to show point values and interecept clicks on the plot
	
	var options = Function.new( # Function is a weird name for trace options
		x, y, "Pressure", 
		{ 
			color = Color("#36a2eb"),
			marker = Function.Marker.CIRCLE,
			type = Function.Type.LINE,
			interpolation = Function.Interpolation.STAIR
		}
	)
	
	chart.plot([options], properties)

```

The setup is available here: https://github.com/chaojian-zhang/Sample_Plotting_with_EasyChart

**Comment by fenix-hub** [comment]
Hi, and welcome.
Thank you for your effort!
I'm sorry about the lack of documentation.
I'm not able to provide one at the moment due to personal business and the fact that this is still a "work in progress" project. So any suggestion is always welcomed.
I'll make sure to start recording a couple of videos as a getting started guide.

**Comment by chaojian-zhang** [comment]
Videos would be helpful!
Thanks again for sharing!

**Comment by fenix-hub** [comment]
you are welcome!
thank you.

**Comment by namelessvoid** [comment]
@chaojian-zhang @fenix-hub Since I had the same struggle at the beginning, I spent some time on putting together a short getting started guide on the wiki on my fork. I want to add some more content about theming and maybe giving an overview on the different examples included.

You find it here: https://github.com/namelessvoid/godot-engine.easy-charts/wiki

Feel free to have a look and provide feedback to me!

**Comment by namelessvoid** [comment]
@fenix-hub I "finished" the documentation with the (I think) most important points: Install, Getting started, Theming and some advanced topics (dynamic data and discrete domains). Please have a look at https://github.com/namelessvoid/godot-engine.easy-charts/wiki and let me know if you think that this could be a good enough starting point.

Updating the wiki is not as easy, because I cannot create a PR for the wiki repo. You have to clone the wiki from my repo and copy the content into your wiki repo, I think.

**Comment by namelessvoid** [comment]
@fenix-hub I was thinking if it makes more sense to incorporate the docs into a `docs/` directory and remove the wiki. This would simplify the workflow for you as well as for contributors.

**Comment by fenix-hub** [comment]
@namelessvoid Yes, I agree with you. I was thinking on using Mkdocs since it is easy to use and collaborate on, supports markdown (so all the wiki pages can just be copy-pasted and then improved), and is easy to setup with Github Pages and Github actions. What do you think about it?

**Comment by namelessvoid** [comment]
Thanks for your input, @fenix-hub! I hope you had a great weekend :)

I did compare Sphinx (which is used by godot itself and by default uses re-structured text) and MkDocs. Although Sphinx is more powerful, I agree that MkDocs is the easier tool to get started, esp. since markdown is more common than re-structured text. Similar to Sphinx, MkDocs is compatible with ReadTheDocs, too. And, if we need any of the advanced features of Sphinx later, we can still switch and use MyST to integrate markdown files.

Long story short, I created a PR #141 that moves the documentation into the repository using MkDocs. It also features a ReadTheDocs integration that you can see live on that URL: https://namelessvoid-godot-engineeasy-charts.readthedocs.io/en/latest/

The ReadTheDocs integration is quite easy (did this for the first time today and it took me only half an hour, including reading :D). The getting started guide can be found here: https://docs.readthedocs.com/platform/stable/tutorial/index.html
